handle.go

```diff
@@ -65,22 +65,37 @@ func NewHandle(rules any, keyObfuscatorRegex string, valueObfuscatorRegex string
 
 	config := newConfig(&encoder.cgoRefs, keyObfuscatorRegex, valueObfuscatorRegex)
 	diagnosticsWafObj := new(wafObject)
+	defer func() {
+		if !diagnosticsWafObj.isInvalid() {
+			wafLib.wafObjectFree(diagnosticsWafObj)
+		}
```

Comment:
Nit: create a wafObjectFree helper doing this check. I also think the WAF already handles the invalid condition.